src/client_side.cc

```diff
@@ -4103,13 +4122,22 @@ ConnStateData::checkLogging()
 bool
 ConnStateData::mayTunnelUnsupportedProto()
 {
-    return Config.accessList.on_unsupported_protocol
+    if (!Config.accessList.on_unsupported_protocol)
+        return false;
+
 #if USE_OPENSSL
-           &&
-           ((port->flags.isIntercepted() && port->flags.tunnelSslBumping)
-            || (serverBump() && pinning.serverConnection))
+    // SslBump processing:
+    //  1) parsing the TLS client handshake
+    //  2) reading and processing the first bumped HTTP request
+    if (sslBumpMode != Ssl::bumpEnd && parsedBumpedRequestCount <= 1)
+        return true;
 #endif
-           ;
+
+    // the first request in a connection to a plain intercepting port
+    if (transparent() && pipeline.nrequests <= 1)
```

Comment:
AFAICT, transparent() does not imply "plain" mentioned in your comment. If you want to exclude TLS connections, then you need more checks.

Reply:
> The TLS connections in these cases should handled (and always return true) in previous checks.

I see no evidence of that. AFAICT, the only "previous check" that returns in this method uses the following condition:

```
USE_OPENSSL && sslBumpMode != Ssl::bumpEnd && parsedBumpedRequestCount <= 1
```

How could that possibly handle all transparent() TLS connections?! The trivial counter-example is `parsedBumpedRequestCount == 100`.

I will resolve this change request here but will reopen it in the new review because this problem has not been addressed.